tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rewriter2/package.scala

```diff
@@ -0,0 +1,11 @@
+package at.forsyte.apalache.tla.bmcmt
+
+import scalaz._
+import scalaz.Scalaz._
+
+package object rewriter2 {
+
+  type RewritingComputation[T] = State[RewritingState, T]
+
+  type StateCompWithExceptions[T] = EitherT[Exception, RewritingComputation, T]
```

Comment:
How about something like this:

```suggestion
  type RewritingComputationExn[T] = EitherT[Exception, RewritingComputation, T]
```

Otherwise its not clear from the name that this pertains to the `RewritingComputation` or that `Comp` is short for `Computation` rather than `Comparison`.